File: go/739-Daily-Temperatures.go
package leetcode

func dailyTemperatures(temperatures []int) []int {

	result := make([]int, 0)

	for i := 0; i < len(temperatures); i++ {
		for j := i; j < len(temperatures); j++ {
			if temperatures[i] < temperatures[j] {
				result = append(result, j-i)
				break
			}

			if j == len(temperatures)-1 {
				result = append(result, 0)
			}
		}
	}

	return result
}

File: csharp/739-Daily-Temperatures.cs
public class Solution
{
    public int[] DailyTemperatures(int[] temperatures)
    {
        var result = new int[temperatures.Length];
        Array.Fill(result, 0);
        var stack = new Stack<int>();

        for (var i = 0; i < temperatures.Length; i++)
        {
            var t = temperatures[i];
            while (stack.Any() && temperatures[stack.Peek()] < t)
            {
                var prev = stack.Pop();
                result[prev] = i - prev;
            }
            stack.Push(i);
        }

        return result;
    }
}